improver/visibility/visibility_combine_cloud_base.py

```diff
@@ -0,0 +1,221 @@
+        )
+
+        return scaling_factors
+
+    def process(self, cubes: CubeList) -> Cube:
+        """
+        Combines the visibility cube with the cloud base at ground level cube. This is
+        done for every visibility threshold independently. Before combining, the cloud
+        base at ground level cube is multiplied by a scaling factor which either reduces
+        or has no impact on the probabilities in the cube. The scaling factor is calculated
+        based upon which visibility threshold the cloud base will be combined with.
+        The maximum probability is then taken between the scaled cloud base cube and the
+        corresponding visibility threshold.
+
+        Args:
+
+            cubes (iris.cube.CubeList):
+                containing:
+                    visibility (iris.cube.Cube):
+                        Cube of probability of visibility relative to thresholds
+                    cloud base at ground level (iris.cube.Cube):
+                        Cube of probability of cloud base at ground level. This cube should only
+                        have spatial dimensions (e.g. spot_index or x,y coordinates).
+        Returns:
+            Cube of visibility data that has been combined with the scaled cloud base at ground
+            level cube.
+
+        Raises:
+            ValueError:
+                If visibility_cube is not a probability cube
+            ValueError:
+                If cloud_base_ground_cube is not a probability cube
+        """
+
+        visibility_cube, cloud_base_ground_cube = self.separate_input_cubes(cubes)
+
+        for cube in [visibility_cube, cloud_base_ground_cube]:
+            if not is_probability(cube):
+                raise ValueError(f"Cube {cube.name()} must be a probability cube")
+
+        # Ensure all probabilities are probability of being below threshold and if not invert
+        # the probabilities.
+        vis_inverted = False
+        if probability_is_above_or_below(visibility_cube) == "above":
+            visibility_cube = invert_probabilities(visibility_cube)
+            vis_inverted = True
+
+        # Calculate the scaling factors for each visibility threshold
+        visibility_threshold_coord = find_threshold_coordinate(visibility_cube).name()
+        visibility_thresholds = visibility_cube.coord(visibility_threshold_coord).points
+        cloud_base_scaling_factor = self.get_scaling_factors(visibility_thresholds)
+
+        # Combine each visibility threshold with the scaled cloud base at ground level cube
+        vis_combined_list = CubeList()
+        for vis_slice, scaling_factor in zip(
+            visibility_cube.slices_over(visibility_threshold_coord),
+            cloud_base_scaling_factor,
+        ):
+            scaled_cloud_base = cloud_base_ground_cube * scaling_factor
+            vis_combined_list.append(
+                Combine(operation="max", expand_bound=False)(
+                    CubeList([vis_slice, scaled_cloud_base])
```

Comment:
I don't think it's necessary. The Combine is tidier at managing the metadata and meant I didn't need to realize the data in  the plugin although if the data is being realized in the Combine plugin this might not matter?  It's conveniant to use but if numpy will be faster then I think that would make more sense to use.